HealthyFood/Maze/MazeStuff/Cells/PileOfCoins.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Maze.MazeStuff.Characters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Maze.MazeStuff.Cells
+{
+    public class PileOfCoins : BaseCell
+    {
+        private Random random = new Random();
+
+        public PileOfCoins(int x, int y, MazeLevel level) : base(x, y, level)
+        {
+        }
+
+        public override CellType CellType => CellType.PileOfCoins;
+
+        public override bool TryToStep(ICharacter character)
+        {
+            var randomAmountOfCoins = random.Next(2, 10);
+            character.Coins = character.Coins + randomAmountOfCoins;
```

Comment:
It's ok but there is short way to write the same code
`character.Coins += randomAmountOfCoins;`